src/Swarm/Game/Step.hs

```diff
@@ -968,6 +968,14 @@ execConst c vs s k = do
 
         return $ Out res s k
       _ -> badConst
+    Knows -> case vs of
+      [VString name] -> do
+        inv <- use robotInventory
```

Comment:
We should also check in `installedDevices`, since they are certainly known but there may not be any in the inventory.  (I am coming to regret separating them into two different inventories.)

Reply:
Do you think it would be complicated to move the installed devices to the entity inventory? For example, could we just change the counts map to include a boolean value for the isInstalled status?

Reply:
I don't think it would be complicated. But we would have to be careful keeping track of counts, and carefully scrutinize all current uses of the inventory to see whether they need to be updated. It would mean a big refactoring, touching a lot of code, without the type checker's help.  I don't know if it would really be worth it.